src/EditorFeatures/CSharp/LanguageServer/CSharpLspBuildOnlyDiagnostics.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+
+namespace Microsoft.CodeAnalysis.CSharp.LanguageServer
+{
+    // Keep in sync with IsBuildOnlyDiagnostic
+    // src\Compilers\CSharp\Portable\Errors\ErrorFacts.cs
+    [LspBuildOnlyDiagnostics(
```

Comment:
>LspBuildOnlyDiagnostics [](http://example.com/codeflow?start=5&length=23)

Since these are just strings do we need to MEF export these, instead of just listing all of them (C#+VB) in a single list?

Reply:
Where would that single list live?  I would not want it in a non-C#/VB location as that would not be scalable to other languages.  it would also be leaking specific compiler lang details into a core location.  THis approach keeps C# in C# and VB in VB (And F# in F# if they ever move onto us, for example).  :)

Reply:
OK, if the intention is to make this available to F# then this is a good approach.

Reply:
If you wanted it in a single list, it'd be shared at the same level as your LSP server. Since VB & C# use the same server instance/config, that code would store the list. If F# shared the same server, it would add it's items to the VB/C# list. If F# used a separate LSP server, it'd manage its own list.